lib/spidr/page/sitemap.rb

```diff
@@ -0,0 +1,194 @@
+require 'nokogiri'
+require 'zlib'
+
+module Spidr
+  class Page
```

Comment:
I feel like there should be a `Sitemap` class that inherits from `Page`. Only the `sitemap.xml` file should have sitemap methods. This would also allow for these methods to not contain `sitemap` in them, which seems kind of redundant if we're already parsing a `sitemap.xml` file.